src/executor/insert.rs

```diff
@@ -137,29 +137,38 @@ where
 {
     type PrimaryKey<A> = <<A as ActiveModelTrait>::Entity as EntityTrait>::PrimaryKey;
     type ValueTypeOf<A> = <PrimaryKey<A> as PrimaryKeyTrait>::ValueType;
-    let last_insert_id_opt = match db.support_returning() {
-        true => {
-            let cols = PrimaryKey::<A>::iter()
-                .map(|col| col.to_string())
-                .collect::<Vec<_>>();
-            let rows = db.query_all(statement).await?;
-            let res = rows.last().ok_or_else(|| {
-                DbErr::RecordNotInserted("None of the records are being inserted".to_owned())
-            })?;
-            res.try_get_many("", cols.as_ref()).ok()
+
+    let last_insert_id = if db.support_returning() {
```

Comment:
That's much better! Do we absolutely care about `RecordNotInserted` if the primary key is provided? How can we detect it if the db doesn't support returning?

If we don't especially care about it, we could rewrite it as : 
```rust
match (primary_key, db.support_returning()) {
  (Some(value_tuple), _) => {db.execute(); Ok(FromValueTuple)},
  (None, true) =>{db.query_all() ... },
  (None, false) => {res = db.execute(); res.last_insert_id()},
}
```

WDYT?

Reply:
Yes, I think we care. Because that's the case when we inserted a duplicate record?

> How can we detect it if the db doesn't support returning

Now it's a compile-time hardcoded value, with SQLite being false. Ideally, SQLite support should be determined by its version detected runtime.

Reply:
No, I mean how can we detect that the record was not inserted if the db doesn't support returning?

Reply:
Umm, I see what you mean

Reply:
Could we use https://stackoverflow.com/questions/6551903/getting-the-number-of-affected-rows-for-a-sqlite-statement-using-the-c-api ?

Reply:
For sqlite, by calling `sqlite3_changes()` (or by selecting `changes()` in the query), we can get the number of rows affected. With that, we can see if it's 0 (didn't insert anything) or >0 (inserted at least one thing). That wouldn't catch the case where we insert >1 value but only some failed, but for single value inserts, it would already be helpful.

Reply:
I think the code as of "Refactor" can be released in 0.10 and further changes should go into 0.11

Reply:
Sounds good. Just FYI, I tested `select changes();` on SQlite3, it seems to do what we want: after a single insert, it'll return 1 if the insert was successful, and 0 otherwise. And it's scoped to the DB connection as well, so it should be usable even in async contexts.

But anyway, inserting a single duplicate row returns an error, so probably the `await?` should trigger a return, no?

Reply:
Yes. It should be a UNIQUE CONSTRAINT ERROR unless if we specified ON CONFLICT DO NOTHING.